fileuploader-kiwiirc-plugin/components/SidebarFileList.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+    <div class="kiwi-filebuffer-outer-container">
+        <div v-if="!fileList.length" class="kiwi-filebuffer-status-message">
+            No files have recently been <br>
+            uploaded to this channel {{currentBuffer}}
+        </div>
+        <div v-else class="kiwi-filebuffer-inner-container">
+            <span class="kiwi-filebuffer-status-message">Files recently uploaded to {{currentBuffer}}</span><br><br>
+            <table class="kiwi-filebuffer-table">
+                <tr><th>Nick</th><th>Link</th><th>Time</th></tr>
+                <tr v-for="(file, idx) in fileList" :key="file">
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.nick}}
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        <a
+                            :href="file.url"
+                            @click.prevent.stop="loadContent(file.url)"
+                            class="kiwi-filebuffer-anchor"
+                        >
+                            {{getFileName(file.url)}}
+                        </a>
+                    </td>
+                    <td class="kiwi-filebuffer-table-td kiwi-filebuffer-time">
+                        {{file.time}}
+                    </td>
+                </tr>
+            </table>
+        </div>
+    </div>
+</template>
+
+<script>
+
+'kiwi public';
+
+export default {
+    data() {
+        return {
+            settings: kiwi.state.setting('fileuploader'),
+        };
+    },
+    methods: {
+        getFileName(file) {
+            file = decodeURI(file);
+            let name = file.split('/')[file.split('/').length-1];
+            if (name.length >= 25) {
+                name = name.substring(0, 18) + '...' + name.substring(name.length - 4);
+            }
+            return name;
+        },
+        loadContent(url) {
+            kiwi.emit('mediaviewer.show', url);
+        },
+        messageHandler() {
+            setTimeout(() => this.sharedFiles(kiwi.state.getActiveBuffer()), 1100)
+        },
+        sharedFiles(buffer) {
+            let returnArr = []
+            let messages = buffer.getMessages()
+            let tmp = buffer.message_count
+            for(let i = 0; i < messages.length; i++) {
+                let e = messages[i]
+                if (e.message.indexOf(this.settings.server) !== -1) {
+                    let time = new Intl.DateTimeFormat('default', { hour: 'numeric', minute: 'numeric', second: 'numeric' }).format(new Date(e.time))
+                    let link = {
+                        url: e.message.substring(e.message.indexOf(this.settings.server)),
```

Comment:
If we add a `filanemae` property and only call the `getFilename()` method here,  then plugins tapping into the `files.listshared` event can use whatever format they wish when adding files to the file array

Reply:
done.